### PR TITLE
Fix params

### PR DESCRIPTION
**Pull request by davvil**
Added parameter --fix-params for fixing parameters during training. If approved this supersedes #314.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [ ] Were system tests modified? If so did you run these at least 5 times to account for the variation across runs?
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/model.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def __init__(self,
                  config_loss: loss.LossConfig,
                  weight_tying: bool = False,
                  weight_tying_type: Optional[str] = C.WEIGHT_TYING_TRG_SOFTMAX,
-                 weight_normalization: bool = False) -> None:
+                 weight_normalization: bool = False):
```
Comment (fhieber) [inline sockeye/model.py]: "-> None" missing

File: sockeye/arguments.py [hunk 2]
```diff
@@ -895,6 +895,11 @@ def add_training_args(params):
                               choices=[C.RNN_INIT_ORTHOGONAL, C.RNN_INIT_ORTHOGONAL_STACKED, C.RNN_INIT_DEFAULT],
                               help="Initialization method for RNN parameters. Default: %(default)s.")
 
+    train_params.add_argument('--fix-params',
```
Comment (fhieber) [inline sockeye/arguments.py]: I would vote for calling this "fixed-params" or even "fixed-param-names", which is consistent with what MXNet expects.

File: sockeye/train.py [hunk 3]
```diff
@@ -640,7 +640,8 @@ def create_training_model(config: model.ModelConfig,
                                             provide_label=train_iter.provide_label,
                                             default_bucket_key=train_iter.default_bucket_key,
                                             bucketing=not args.no_bucketing,
-                                            gradient_compression_params=gradient_compression_params(args))
+                                            gradient_compression_params=gradient_compression_params(args),
+                                            fixed_params=args.fix_params)
```
Comment (fhieber) [inline sockeye/train.py]: I would call both the value and the parameter "fixed_param_names"

File: sockeye/arguments.py (hunk 2 above)
Comment (davvil) [reply sockeye/arguments.py]: I was tempted to do so as well, but I have the feeling that most of our arguments (if appropriate) refer to actions: --restrict-lexicon, --fill-up, --use-tensorboard, etc.

Comment (fhieber) [reply sockeye/arguments.py]: True, but to me `--fix-params` sounds like a boolean flag (fix params if this flag is given) and doesn't encourage me to put a list of names after it).

**Pull request merged**